File: Lista de Exercicio Paradigmas/Lista Java/Ex14.java
import java.util.HashMap;
import java.util.Map;

class Configuracao {
    private static Configuracao instancia = null;
    private Map<String, String> configuracoes;

    private Configuracao() {
        configuracoes = new HashMap<>();
    }

    public static Configuracao getInstancia() {
        if (instancia == null) {
            instancia = new Configuracao();
        }
        return instancia;
    }

    public void set(String chave, String valor) {
        configuracoes.put(chave, valor);
    }

    public String get(String chave) {
        return configuracoes.get(chave);
    }

    public static void main(String[] args) {
        Configuracao config1 = Configuracao.getInstancia();
        config1.set("url", "https://exemplo.com");

        Configuracao config2 = Configuracao.getInstancia();
        System.out.println(config2.get("url"));

        System.out.println(config1 == config2);
    }
}

File: Lista de Exercicio Paradigmas/Lista Python/Ex14.py
class Configuracao:
    _instancia = None

    def __new__(cls, *args, **kwargs):
        if not cls._instancia:
            cls._instancia = super(Configuracao, cls).__new__(cls)
            cls._instancia.configuracoes = {}
        return cls._instancia

    def set(self, chave, valor):
        self.configuracoes[chave] = valor

    def get(self, chave):
        return self.configuracoes.get(chave, None)

config1 = Configuracao()
config1.set("url", "https://exemplo.com")

config2 = Configuracao()
print(config2.get("url"))  

print(config1 is config2)  